### PR TITLE
Alter, delete, update table contents

### PR DESCRIPTION
**Pull request by Meri-MG**
In this PR was: 
- Used database transactions;
- Modified and deleted data in SQL.
- Prepared complex queries that answer analytical questions.

### CODE REVIEW

File: queries.sql [hunk 1]
```diff
@@ -6,3 +6,57 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SET species = 'digimon'
+WHERE name LIKE '%mon';
+BEGIN;
+UPDATE animals 
+SET species = 'pokemon'
+WHERE name NOT LIKE '%mon';
+COMMIT;
+BEGIN;
+DELETE FROM animals;
+ROLLBACK;
```
Comment (DammyShittu) [inline queries.sql]: - Kindly add steps to verify that changes were made at particular points as the requirement states. 



For Example 👇 

```
   BEGIN;
   UPDATE animals
   SET species = 'unspecified';
   SELECT species from animals;
   ROLLBACK;
   SELECT species from animals;
```

File: queries.sql [hunk 2]
```diff
@@ -6,3 +6,57 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+ROLLBACK TO SAVEPOINT del1;
+BEGIN;
+UPDATE animals
+SET weight_kg = weight_kg * (-1)
+WHERE weight_kg < 0;
+COMMIT;
+-- Write queries to answer the questions
+SELECT COUNT(*) FROM animals; //10
+SELECT COUNT(escape_attempts) FROM animals
+WHERE neutered = false; //3
```
Comment (DammyShittu) [inline queries.sql]: - I believe the query here says `How many animals have never tried to escape?` To do this, you can `COUNT` all animals where `escape_attempts` is equal to zero. 😉

File: queries.sql [hunk 3]
```diff
@@ -6,3 +6,57 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT AVG(weight_kg) FROM animals; //15.55...
+SELECT SUM(escape_attempts) FROM animals
+WHERE neutered = false; // 4
+SELECT SUM(escape_attempts) FROM animals
+WHERE neutered = true; // 20
```
Comment (DammyShittu) [inline queries.sql]: - It's nice that you tried to do these separately to know which animals escape the most. 👍 However, you can also do it all in one query. 😀 
```sql
   SELECT neutered, AVG(escape_attempts) FROM animals
   GROUP BY neutered;
```

File: queries.sql [hunk 4]
```diff
@@ -6,3 +6,57 @@ SELECT name, escape_attempts FROM animals WHERE weight_kg > 10.5;
...
+SELECT MAX(weight_kg) FROM animals
+WHERE species = 'digimon'; // 45
+SELECT MIN(weight_kg) FROM animals
+WHERE species = 'digimon'; // 5.7
+SELECT MAX(weight_kg) FROM animals
+WHERE species = 'pokemon'; // 17
+SELECT MIN(weight_kg) FROM animals
+WHERE species = 'pokemon'; // 11
```
Comment (DammyShittu) [inline queries.sql]: - What do you think about saving about five lines of code to do these in a query? 
Here 👇 is an example of how to write this query

```sql
SELECT species, MIN(weight_kg), MAX(weight_kg) FROM animals
   GROUP BY species;
```

**Review by DammyShittu — CHANGES_REQUESTED** [review]
Hi @Meri-MG, 

Great job so far! 👏 

There are some issues that you still need to work on to go to the next milestone but you are almost there!

## Required Changes ♻️

- Writing individual SQL queries is amazing. However, a lot of times, especially when working with identical data, the `GROUP BY` clause is a really helpful addition as it makes it easy to query records with matching/identical columns. For more details on how to use the `GROUP BY` clause, please, go through this [guide](https://www.tutorialspoint.com/sql/sql-group-by.htm). 👍 

_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:

- adding linebreaks between data inputs and queries to enhance readability. 😉 

## Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by addod19 — APPROVED** [review]
### Hi @Meri-MG, 

Your project is complete!  Nice job fix all required changes :fireworks: There is nothing else to say other than... it's time to merge it :shipit: 
Congratulations! 🎉 



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.


------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**